File: app/src/main/java/org/worshipsongs/utils/ImageUtils.java
package org.worshipsongs.utils;

import android.content.res.Resources;
import android.graphics.Bitmap;
import android.graphics.drawable.BitmapDrawable;
import android.graphics.drawable.Drawable;

/**
 * Author : Madasamy
 * Version : x.x.x
 */

public class ImageUtils
{

    public static Bitmap resizeBitmapImageFn(Bitmap bitmapSource, int maxResolution)
    {
        int iWidth = bitmapSource.getWidth();
        int iHeight = bitmapSource.getHeight();
        int newWidth = iWidth;
        int newHeight = iHeight;
        float rate = 0.0f;

        if (iWidth > iHeight) {
            if (maxResolution < iWidth) {
                rate = maxResolution / (float) iWidth;
                newHeight = (int) (iHeight * rate);
                newWidth = maxResolution;
            }
        } else {
            if (maxResolution < iHeight) {
                rate = maxResolution / (float) iHeight;
                newWidth = (int) (iWidth * rate);
                newHeight = maxResolution;
            }
        }
        return Bitmap.createScaledBitmap(bitmapSource, newWidth, newHeight, true);
    }

    public static Drawable resizeBitmapImageFn(Resources resources, Bitmap bitmapSource, int maxResolution)
    {
        int iWidth = bitmapSource.getWidth();
        int iHeight = bitmapSource.getHeight();
        int newWidth = iWidth;
        int newHeight = iHeight;
        float rate = 0.0f;

        if (iWidth > iHeight) {
            if (maxResolution < iWidth) {
                rate = maxResolution / (float) iWidth;
                newHeight = (int) (iHeight * rate);
                newWidth = maxResolution;
            }
        } else {
            if (maxResolution < iHeight) {
                rate = maxResolution / (float) iHeight;
                newWidth = (int) (iWidth * rate);
                newHeight = maxResolution;
            }
        }
        return new BitmapDrawable(resources, Bitmap.createScaledBitmap(bitmapSource, newWidth, newHeight, true));
    }
}

File: app/src/main/java/org/worshipsongs/utils/ImageUtils.kt
package org.worshipsongs.utils

import android.content.res.Resources
import android.graphics.Bitmap
import android.graphics.drawable.BitmapDrawable
import android.graphics.drawable.Drawable

/**
 * Author : Madasamy
 * Version : x.x.x
 */

object ImageUtils
{

    fun resizeBitmapImageFn(bitmapSource: Bitmap, maxResolution: Int): Bitmap
    {
        val iWidth = bitmapSource.width
        val iHeight = bitmapSource.height
        var newWidth = iWidth
        var newHeight = iHeight
        var rate = 0.0f

        if (iWidth > iHeight)
        {
            if (maxResolution < iWidth)
            {
                rate = maxResolution / iWidth.toFloat()
                newHeight = (iHeight * rate).toInt()
                newWidth = maxResolution
            }
        } else
        {
            if (maxResolution < iHeight)
            {
                rate = maxResolution / iHeight.toFloat()
                newWidth = (iWidth * rate).toInt()
                newHeight = maxResolution
            }
        }
        return Bitmap.createScaledBitmap(bitmapSource, newWidth, newHeight, true)
    }

    fun resizeBitmapImageFn(resources: Resources, bitmapSource: Bitmap, maxResolution: Int): Drawable
    {
        val iWidth = bitmapSource.width
        val iHeight = bitmapSource.height
        var newWidth = iWidth
        var newHeight = iHeight
        var rate = 0.0f

        if (iWidth > iHeight)
        {
            if (maxResolution < iWidth)
            {
                rate = maxResolution / iWidth.toFloat()
                newHeight = (iHeight * rate).toInt()
                newWidth = maxResolution
            }
        } else
        {
            if (maxResolution < iHeight)
            {
                rate = maxResolution / iHeight.toFloat()
                newWidth = (iWidth * rate).toInt()
                newHeight = maxResolution
            }
        }
        return BitmapDrawable(resources, Bitmap.createScaledBitmap(bitmapSource, newWidth, newHeight, true))
    }
}
